lib/scripts/diffExperimentBuilds.py

```diff
@@ -0,0 +1,262 @@
+import argparse
+import os
+import subprocess
+import sys
+import shutil
+
+import benchpark.paths
+
+sys.path.append(str(benchpark.paths.benchpark_home) + "/spack/lib/spack")
+import llnl.util.tty.color as color  # noqa: E402
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="""Compare YAMLs generated from 'system init' between two different commits of Benchpark.
+        (check if the output of system init was changed between commits).
+
+        Examples:
+            - benchpark-python diffExperimentBuilds.py -s llnl-cluster -c ruby -a cpu
+            - benchpark-python diffExperimentBuilds.py -s llnl-sierra -a cuda
+            - benchpark-python diffExperimentBuilds.py -s llnl-elcapitan -c tioga -a rocm""",
+        formatter_class=argparse.RawTextHelpFormatter,
+    )
+    parser.add_argument(
+        "-o",
+        "--old",
+        type=str,
+        default="develop",
+        help="Commit hash or branch name for the old version of Benchpark (default: develop).",
+    )
+    parser.add_argument(
+        "-n",
+        "--new",
+        type=str,
+        default="pearce8-patch-3",
+        help="Commit hash or branch name for the new version of Benchpark",
+    )
+    parser.add_argument(
+        "-s",
+        "--system",
+        type=str,
+        required=True,
+        help="System being ran on. ex: 'llnl-cluster'",
+    )
+    parser.add_argument(
+        "-c",
+        "--cluster",
+        type=str,
+        default=None,
+        help="cluster variant if applicable. ex: 'ruby'",
+    )
+    parser.add_argument(
+        "-a",
+        "--arch",
+        type=str,
+        required=True,
+        help="Architecture for experiment set. choose from: ('cpu', 'cuda', 'rocm')",
+    )
+    parser.add_argument(
+        "--extra-spec",
+        default="",
+        type=str,
+        help="provide additional spec to experiment spec string",
+    )
+    parser.add_argument(
+        "-e",
+        "--experiments",
+        nargs="*",
+        default="",
+    )
+    parser.add_argument("--run-experiment", action="store_true")
+
+    args = parser.parse_args()
+
+    old_name = f"benchpark-old-{args.arch}"
+    new_name = f"benchpark-new-{args.arch}"
+    # Use the arguments provided by the user or the defaults
+    bp = {
+        old_name: args.old,
+        new_name: args.new,
+    }
+
+    experiments_dict = {
```

Comment:
Please replace the static dict with a dynamic list generated by #486.

Reply:
Done